imitation/util/trainer.py

```diff
@@ -56,6 +53,33 @@ def init_trainer(env_id, use_gail=False,
                               scale=discrim_scale, **reward_kwargs)
     discrim = discrim_net.DiscrimNetAIRL(rn, **discrim_kwargs)
 
-  trainer = Trainer(env, gen_policy, discrim,
-                    expert_policies=expert_policies, **trainer_kwargs)
+  ro_paths = util.get_dump_paths(
+    env,
+    basedir=rollouts_dir,
+    policy_class=gen_policy.__class__,
```

Comment:
Why do we need the policy the demonstrations were generated by be the same as the policy we are using in IRL training?

Reply:
My suggestion would be to just make trainer take a path (or, if we want to load multiple rollouts, a glob) to the file containing rollouts. This is simple, does not couple `train` to `data_collect` and supports a wide variety of use cases. Then no need to worry about the generator policy at all.

The magic of `trainer` discovering `data_collect`'s output is convenient in the simple use case, but it can easily produce unexpected results and is inflexible. It makes it hard for a user to use something other than `data_collect`: they'd have to make the filenames match exactly what's expected. Even if they are using `data_collect`, what if they've trained multiple different policies?

Reply:
Ended up choosing globs. Thanks for your input!